finality-aleph/src/sync/service.rs

```diff
@@ -282,10 +289,11 @@ where
             Request(request) => {
                 let state = request.state().clone();
                 self.handle_request(request, peer.clone());
-                self.handle_state(state, peer)
+                self.handle_state(state, peer);
             }
-            RequestResponse(_, justifications) => {
-                self.handle_justifications(justifications, Some(peer))
+            RequestResponse(blocks, justifications) => {
+                self.handle_justifications(justifications, Some(peer));
+                self.handle_blocks(blocks);
```

Comment:
hmmm, shouldnt we first import block and only then handle the justifications? Also its quite confusing to me why the response has both blocks and justifications in body. Shouldn't it be the separate response for justifications and blocks?

Reply:
The order doesn't matter that much, as handling justifications only puts them in the forest, they get actually justified when we get info that blocks got imported.

Reply:
also, the main idea is that the response represents "all what can be useful to us according to the peer, based on info from our request", hence single message variant

Reply:
Imho its better for code readability to make distinction here (in enum variant). Anyway resolving, looks good as it is :)